yt_dlp/extractor/youtube.py

```diff
@@ -3403,11 +3435,18 @@ def _real_extract(self, url):
         microformats = traverse_obj(
             player_responses, (..., 'microformat', 'playerMicroformatRenderer'),
             expected_type=dict, default=[])
-        video_title = (
-            get_first(video_details, 'title')
-            or self._get_text(microformats, (..., 'title'))
-            or search_meta(['og:title', 'twitter:title', 'title']))
-        video_description = get_first(video_details, 'shortDescription')
+
+        original_title = get_first(video_details, 'title')
+        original_description = get_first(video_details, 'shortDescription')
+        translated_title = self._get_text(microformats, (..., 'title'))
+        translated_description = self._get_text(microformats, (..., 'description'))
+        video_title = search_meta(['og:title', 'twitter:title', 'title'])
+        if self._preferred_lang is not None:
+            video_title = translated_title or original_title or video_title
+            video_description = translated_description or original_description
+        else:
+            video_title = original_title or translated_title or video_title
+            video_description = original_description or translated_description
```

Comment:
I'd write it like this:
```suggestion
        translated_title = self._get_text(microformats, (..., 'title'))
        video_title = (self._preferred_lang and translated_title
                       or get_first(video_details, 'title')
                       or translated_title
                       or search_meta(['og:title', 'twitter:title', 'title']))
        translated_description = self._get_text(microformats, (..., 'description'))
        video_description = (self._preferred_lang and translated_description
                             or get_first(video_details, 'shortDescription')
                             or translated_description)
```

but it's just a matter of personal preference